[New product] Apache Pulsar (#6734)
captn3m0/endoflife.date: products/apache-pulsar.md

---
title: Apache Pulsar
category: server-app
tags: apache java-runtime
iconSlug: apachepulsar
permalink: /apache-pulsar
alternate_urls:
-   /pulsar
releasePolicyLink: https://pulsar.apache.org/contribute/release-policy/

> releaseDateColumn: true

Reviewer: ```suggestion
```

This is the default.

eolColumn: Active Support
eoesColumn: Security Support

auto:
  methods:
  -   git: https://github.com/apache/pulsar.git
  -   release_table: https://pulsar.apache.org/contribute/release-policy/
      selector: ".MuiTable-root"
      fields:
        releaseCycle:
          column: "Version"
          regex: '(?P<value>\d+\.\d+)\s?.*'
        eol:
          column: "Active Support"
          regex: '.*\((?P<value>.*)\)'
        eoes:
          column: "Security Support"

>           regex: '.*\((?P<value>.*)\)'

Reviewer: ```suggestion
auto:
  methods:
  -   github_releases: apache/pulsar
  -   release_table: https://pulsar.apache.org/contribute/release-policy/
      selector: "table"
      fields:
        releaseCycle:
          column: "Version"
          regex: '(?P<value>\d+\.\d+).*'
        releaseDate:
          column: "Released"
          regex: '.*\((?P<value>.+)\)'
        eol:
          column: "Active Support"
          regex: '.*\((?P<value>.+)\)'
        eoes:
          column: "Security Support"
          regex: '.*\((?P<value>.+)\)'
```

For more accurate dates. This should resolve the point you noted:

> One detail I noticed was that the Git tags was dated earlier than the release date on both the release policy and release notes pages. See for example 2.10. I didn't figure out how to fix that.

-   releaseCycle: "2.11"
    releaseDate: 2023-01-11
    eol: 2024-01-11
    eoes: 2024-01-11
    latest: '2.11.4'
    latestReleaseDate: 2024-03-08

-   releaseCycle: "2.10"
    releaseDate: 2022-04-18
    eol: 2023-04-18
    eoes: 2023-04-18
    latest: '2.10.6'
    latestReleaseDate: 2024-03-08

-   releaseCycle: "2.9"
    releaseDate: 2021-12-20
    eol: 2022-12-20
    eoes: 2022-12-20
    latest: '2.9.5'
    latestReleaseDate: 2023-04-20

-   releaseCycle: "2.8"
    releaseDate: 2021-06-15
    eol: 2022-06-15
    eoes: 2022-06-15
    latest: '2.8.4'
    latestReleaseDate: 2022-10-04

-   releaseCycle: "2.7"
    releaseDate: 2020-12-03
    eol: 2021-12-03
    eoes: 2021-12-03
    latest: '2.7.5'
    latestReleaseDate: 2022-09-06

-   releaseCycle: "2.6"
    releaseDate: 2020-06-17
    eol: 2021-06-17
    eoes: 2021-06-17
    latest: '2.6.4'
    latestReleaseDate: 2021-06-02

-   releaseCycle: "2.5"
    releaseDate: 2020-01-15
    eol: 2021-01-15
    eoes: 2021-01-15
    latest: '2.5.2'
    latestReleaseDate: 2020-05-19

---

> [Apache Pulsar](https://pulsar.apache.org) is an open-source, distributed
> messaging and streaming platform built for the cloud.

> The last two LTS releases and the last two feature releases are supported.
> Security patches are provided for the past three LTS releases and two feature
> releases.
> 
> ## [Versioning](https://pulsar.apache.org/contribute/release-policy/#release-semantics)
> 
> The Pulsar project follows a variant of Semantic Versioning (semver), replacing
> `major.minor.patch` with `LTS.feature.patch`.
> 
> Concretely, existing releases can expect patches for bugs and security
> vulnerabilities. New features will target feature releases. A "major" version
> bump will not carry any special meaning in terms of "big features" included in
> the release or breaking API changes. Instead, it will simply signal a new
> long-term support (LTS) release.

Reviewer: ```suggestion
The Pulsar project follows [a variant of Semantic Versioning (semver)](https://pulsar.apache.org/contribute/release-policy/#release-semantics),
replacing `major.minor.patch` with `LTS.feature.patch`. With this numbering system:

- a "major" version bump signals a new long-term support (LTS) release,
- a "minor" version bump signals a new feature release, and
- a "patch" version bump signals patches for bugs and security vulnerabilities.

Feature releases are released every 3 months. They are supported with bug and security fixes for 6 months.

LTS releases are released every 18 months. They are supported with bug and security fixes for 24 months,
and with only security fixes for an additional 12 months.

Upgrades should be done linearly [as explained in Apache Pulsar documentation](https://pulsar.apache.org/contribute/release-policy/#compatibility-between-releases).
```

Before talking about LTS, what are the different kind of versions and the release frequency should be detailed.

WDYT of this ?

Reviewer 2: LGTM : could it be merged then work on LTS , see [Apache Pulsar 4.0 LTS since Pulsar 3.0 LTS?](https://pulsar.apache.org/blog/2024/10/24/announcing-apache-pulsar-4-0/) ?

Reviewer: > could it be merged then work on LTS

My comment may be misleading, but the description I gave in my suggestion is already talking about LTS.

And no we won't merge before the review has been finished. And given the limited time I spent on this project I can't give you an ETA.
